internal/uploader/upload.go

```diff
@@ -61,7 +63,7 @@ func (um Uploader) Upload(force bool) error {
 			} else if err != nil {
 				slog.Warn("Failed to upload report", "file", r.Name, "source", um.source, "error", err)
 				mu.Lock()
-				uploadError = errors.Join(uploadError, fmt.Errorf("%s upload failed for report %s: %v", um.source, r.Name, err))
+				uploadError = errors.Join(uploadError, fmt.Errorf("%s upload failed for report %s: %w", um.source, r.Name, err))
 				mu.Unlock()
```

Comment:
I don't see a reason to not call `defer mu.Unlock()` right after `mu.Lock()` in this case. Let's save us from having to think of whether fmt.Errorf or erros.Join could panic or not.